### ISSUE TITLE
Model is not trainable on the example notebook

### ISSUE THREAD

**Issue by fvalle1** [Buffoni/spectral_learning#3]
I am try tu running [Spectral_Layer_example.ipynb](Spectral_Layer_example.ipynb) just straight out of the box, but the model created there is not trainable. I do not think it is expected.

<img width="724" alt="Screenshot 2022-06-30 at 15 59 34" src="https://user-images.githubusercontent.com/7323644/176696693-50c4a104-afd4-4dc8-bf05-598de3a5f963.png">

**Comment by Buffoni** [comment]
Hi Filippo, I'll look into this issue. What version of TF are you running on? Also, this might be caused by not setting neither of the flags "is_base_trainable" and "is_diag_trainable" to True when creating the model. 
I'll also point out that we are currently in the process of discontinuing this repository in favour of a more updated and comprehensive one that you can find here https://github.com/Jamba15/SpectralTools .

**Comment by fvalle1** [comment]
Hi @Buffoni ,
I am using the settings in your notebook `is_diag_trainable` is set to *true*
```python
spectral_config={
    'is_base_trainable' : False,
    'is_diag_start_trainable' : True,
    'is_diag_end_trainable' : True,
    'use_bias' : False
}
```
I am using *tensorflow* *2.9.1* with *Python 3.9.7*.

Thank you for pointing out the new repo, I will surely look into it!

**Comment by Buffoni** [comment]
The TF version might be the problem, the code in this repo is tested to work on version 2.3, for versions greater than 2.4 some things changed and we ourselves observed some bugs. Could you try and run the notebook in an environment with TensorFlow version ==2.3.1 and see if this solves the issue?

**Comment by fvalle1** [comment]
Yes using tensorflow==2.3 solves the issue.
Unfortunately that version is not available on newer version of Python (3.9).

Thank you!

> On 30 Jun 2022, at 17:24, Lorenzo Buffoni ***@***.***> wrote:
> 
> ﻿
> The TF version might be the problem, the code in this repo is tested to work on version 2.3, for versions greater than 2.4 some things changed and we ourselves observed some bugs. Could you try and run the notebook in an environment with TensorFlow version ==2.3.1 and see if this solves the issue?
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you authored the thread.

-- 
------------------------



Indirizzo istituzionale di posta elettronica 
degli studenti e dei laureati dell'Università degli Studi di TorinoOfficial 
University of Turin email address for students and graduates

**Comment by Buffoni** [comment]
Happy that is not a more serious issue, we'll work on forward compatibility in the new repo that I've linked above.
We'll update the README on this repo accordingly soon, for the time being I think this issue can be marked as closed.
